File: .agent/skills/cnma-cap-backend/templates/node/srv/src/domain/__feature__/events/OnAfterUpdate{{FeatureName}}Event.js
import cds from '@sap/cds';

/**
 * OnAfterUpdate{{FeatureName}}Event - AFTER UPDATE event handler.
 * SRP: Handles post-update side effects.
 */
module.exports = {
    execute: async (_, req) => {
        try {
            const userId = req.user?.id || 'anonymous';
            const updatedData = req.data;

            cds.log('{{FeatureName}} updated', {
                userId,
                data: updatedData,
                timestamp: new Date().toISOString()
            });
        } catch (error) {
            cds.log.error('Error in after update {{FeatureName}}: ', error);
        }
    }
};

File: .agent/skills/cnma-cap-backend/templates/java/srv/src/main/java/cnma/{{module_name}}/domain/__feature__/events/OnAfterUpdate{{FeatureName}}Event.java
package cnma.{{module_name}}.domain.__feature__.events;

import com.sap.cds.services.RequestContext;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

/**
 * OnAfterUpdate{{FeatureName}}Event - AFTER UPDATE event handler.
 * SRP: Handles post-update side effects.
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class OnAfterUpdate{{FeatureName}}Event {

    private final RequestContext requestContext;

    public Object execute(Object _, Object req) {
        String userId = getUserId(req);
        log.info("AFTER UPDATE {{FeatureName}} by user: {}", userId);

        // Post-update logic
        return null;
    }

    private String getUserId(Object req) {
        return "anonymous";
    }
}